File: generated-code/example/ruby/model/enumeration.rb
module Enumeration
    VALUE_ONE = 0
    VALUE_TWO = 1

    def self.read_from(stream)
        result = stream.read_int()
        if result < 0 || result >= 2
            raise "Unexpected tag value"
        end
        result
    end

    def self.to_s(value)
        if value == VALUE_ONE
            return "VALUE_ONE"
        end
        if value == VALUE_TWO
            return "VALUE_TWO"
        end
        raise "Impossible happened"
    end

    def self.to_str(value)
        self.to_s(value)
    end
end

File: generated-code/example/cpp/model/Enumeration.cpp
#include "Enumeration.hpp"
#include <stdexcept>

Enumeration readEnumeration(InputStream& stream) {
    switch (stream.readInt()) {
    case 0:
        return Enumeration::VALUE_ONE;
    case 1:
        return Enumeration::VALUE_TWO;
    default:
        throw std::runtime_error("Unexpected tag value");
    }
}

std::string enumerationToString(Enumeration value) {
    switch (value) {
    case Enumeration::VALUE_ONE:
        return "VALUE_ONE";
    case Enumeration::VALUE_TWO:
        return "VALUE_TWO";
    default:
        throw std::runtime_error("Impossible happened");
    }
}